index.js

```diff
@@ -50,12 +50,9 @@ exports.decorateTerms = (Terms, { React }) => {
         );
         if (canvasCollection.length >= 3) {
           const canvases = new Array();
-          document.body.removeEventListener(
-            'DOMSubtreeModified',
-            canvasListener,
-          );
+
           for (let canvas of canvasCollection) canvases.push(canvas);
-          this.setState({ ...this.state, canvases });
+          this.setState({ canvases });
```

Comment:
Try this:
```suggestion
          this.setState(prevState, _ ) => ({...prevState, canvases }) );
```
 I think React doesn't like it when you spread `this.state` inside of `setState`. I think when you do so it throws the app into an infinite loop (maybe?)

Reply:
I think @TejasQ Once told me that updating the state this way with ... was not a good idea and it was unnecessary. That setState only updated the state and it wasn't necessary to explicitly include the old state.

Reply:
Okay That makes sense